File: src/java/magma/app/compile/value/AccessValue.java
package magma.app.compile.value;

import magma.api.option.None;
import magma.api.option.Option;
import magma.api.option.Some;
import magma.app.ValueCompiler;
import magma.app.compile.CompileState;
import magma.app.compile.type.PrimitiveType;
import magma.app.compile.type.Type;

public record AccessValue(Value child, String property) implements Value {
    public String generate() {
        return ValueCompiler.generateValue(this.child) + "." + this.property;
    }

    public Option<Value> toValue() {
        return new Some<Value>(this);
    }

    public Type resolve(CompileState state) {
        return PrimitiveType.Unknown;
    }

    @Override
    public Option<String> generateAsEnumValue(String structureName) {
        return new None<String>();
    }
}

File: src/ts/magma/app/compile/value/AccessValue.ts
import { Value } from "../../../../magma/app/compile/value/Value";
import { Option } from "../../../../magma/api/option/Option";
import { Some } from "../../../../magma/api/option/Some";
export class AccessValue implements Value {
	child: Value;
	property: string;
	constructor (child: Value, property: string) {
		this.child = child;
		this.property = property;
	}
	generate(): string {
		return this.child.generate() + "." + this.property/*unknown*/;
	}
	findChild(): Option<Value> {
		return new Some<Value>(this.child)/*unknown*/;
	}
	is(type: string): boolean {
		return "access".equals(type)/*unknown*/;
	}
}
